cmd/conduit/cecdysis/decorators.go

```diff
@@ -57,24 +60,44 @@ func (CommandWithExecuteWithClientDecorator) Decorate(_ *ecdysis.Ecdysis, cmd *c
 
 		grpcAddress, err := getGRPCAddress(cmd)
 		if err != nil {
-			return fmt.Errorf("error reading gRPC address: %w", err)
+			return cerrors.Errorf("error reading gRPC address: %w", err)
 		}
 
 		client, err := api.NewClient(cmd.Context(), grpcAddress)
 		if err != nil {
-			// Not an error we need to escalate to the main CLI execution. We'll print it out and not execute further.
-			_, _ = fmt.Fprintf(os.Stderr, "%v\n", err)
-			return nil
+			return handleError(err)
 		}
 		defer client.Close()
 
 		ctx := ecdysis.ContextWithCobraCommand(cmd.Context(), cmd)
-		return v.ExecuteWithClient(ctx, client)
+		return handleError(v.ExecuteWithClient(ctx, client))
 	}
 
 	return nil
 }
 
+// check what type of error is to see if it's worth showing `cmd.Usage()` or not.
+// if any error is returned, usage will be shown automatically.
+func handleError(err error) error {
+	errMsg := err.Error()
+	st, ok := status.FromError(err)
+
+	// an API error, we try to parse `desc`
+	if ok {
+		errMsg = st.Message()
+
+		// st.Message() is already an entire representation of the error
+		// need to grab the desc
+		descIndex := strings.Index(errMsg, "desc =")
+		if descIndex != -1 {
+			errMsg = errMsg[descIndex+len("desc = "):]
+		}
+	}
+	_, _ = fmt.Fprintf(os.Stderr, "%v\n", errMsg)
+	return nil
```

Comment:
Wouldn't this mean that the exit code will be 0, i.e. no error? 🤔

Reply:
@lovromazgon that's correct. This is the only way to ensure usage is not shown. The error bubbles up to cobra and if there's an error and we're not silencing usage, this will be shown. I actually just thought of something that I'm going to tru.

Reply:
I'm thinking if we should rather set `SilenceUsage` to `true`, maybe even in ecdysis, so that it's the default (see https://github.com/spf13/cobra/issues/340#issuecomment-243811964). Apparently the usage would still get displayed if `--help` is added or when the subcommand is missing, which is what we're looking for, right? Incorrect CLI usage results in usage, otherwise we silence the usage and display the returned error.

Reply:
Need to investigate further, but the immediate consequence is that usage is not shown on other validations we do make (flags, args)... maybe that's still ok

```
$ conduit pipelines desc 
Error: requires a pipeline ID
exit status 1
```

Reply:
I think I'm ok with this https://github.com/ConduitIO/conduit/pull/2114/commits/c0d321d05295f8df6d0d3b9e8caf39ae7a6e5f87. Much simpler.

Reply:
I see, I also tried this:

```
$ ./conduit foo
Error: unknown command "foo" for "conduit"
Run 'conduit --help' for usage.
```

Not quite what I expected, but I think it's fine 🤷 It's a clear output and a good compromise, given the simplicity of the code change.